docs/JLBP-0016.md

```diff
@@ -8,25 +8,36 @@ permalink: /JLBP-16
 For multi-module projects, this best practice assumes you have already applied
 [JLBP-15](JLBP-0015.md), so that your project has a BOM.
 
-Upper version alignment is defined in the [glossary](glossary.md).
+Upper version alignment means that when more than one version of a dependency
+is present in a module's dependency tree, the build system selects
+the highest version. Upper version alignment ensures that when packages are
+upgraded to higher versions which don’t make any breaking changes,
+there are no new linkage conflicts.
 
 ## Achieving upper version alignment
 
 Upper version alignment increases the likelihood that consumers' build systems
 select the right versions of direct and transitive dependencies, reducing the
 number of conflicts.
 
-As noted in the definition, to fix a version misalignment caused by a shorter
-path to a version of a dependency that is not the upper bound in the
-dependency tree, a direct dependency needs to be added so that Maven consumers
-select the correct version.
+Gradle always chooses the highest version it finds in the dependency graph,
```

Comment:
```suggestion
Gradle by default chooses the highest version it finds in the dependency graph,
```

Gradle can select old versions when it's configured to do so.

Reply:
How would it be so configured?

Reply:
Adding constraints https://docs.gradle.org/current/userguide/java_platform_plugin.html#sec:java_platform_consumption or import BOMs with enforcedPlatform.